src/Workspaces/Core/Portable/Utilities/Contract.cs

```diff
@@ -23,6 +23,13 @@ internal static class Contract
             }
         }
 
+        [return: NotNull]
+        public static T NotNull<T>(T value, string? message = null) where T : class?
```

Comment:
- This is probably a no-go. should use RoslynDebug.Assert added in https://github.com/dotnet/roslyn/pull/38646
- Also we need to check for langversion in SupportsSwitchExpression

Reply:
we do not have access to RoslynDebug, perhaps we should add one for Workspaces?